src/main/java/org/hisp/dhis/rules/functions/RuleFunctionCountIfValue.java

```diff
@@ -84,9 +86,17 @@ private String countIfValue( List<String> arguments, Map<String, RuleVariableVal
 
         RuleVariableValue variableValue = valueMap.get( ruleVariableName );
 
+        String valueToFind = arguments.get(1);
+
         if ( variableValue != null )
         {
-            return Integer.toString( Collections.frequency( variableValue.candidates(), arguments.get( 1 ) ) );
+            if(variableValue.type() == RuleValueType.BOOLEAN){
```

Comment:
Spaces, and new scope should be on new line:
if ( variableValue.type() == RuleValueType.BOOLEAN )
{